From foo@baz Tue Apr  9 12:12:43 2002
Date: Wed, 13 Aug 2008 20:08:35 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB gadget: move f_ecm.c to gadget_ecm_core.ko

As f_ecm.c is included in multiple .ko files, move it into a
standalone gadget_ecm_core.ko

Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Makefile |   12 +++++++-----
 drivers/usb/gadget/f_ecm.c  |   16 ++++++++++++++++
 2 files changed, 23 insertions(+), 5 deletions(-)

--- a/drivers/usb/gadget/f_ecm.c
+++ b/drivers/usb/gadget/f_ecm.c
@@ -734,6 +734,7 @@ fail:
 
 	return status;
 }
+EXPORT_SYMBOL_GPL(ecm_bind);
 
 static void
 ecm_unbind(struct usb_configuration *c, struct usb_function *f)
@@ -829,3 +830,18 @@ int __init ecm_bind_config(struct usb_co
 	}
 	return status;
 }
+EXPORT_SYMBOL_GPL(ecm_bind_config);
+
+static int __init ecm_init(void)
+{
+	return 0;
+}
+
+static void __exit ecm_exit(void)
+{
+}
+
+module_init(ecm_init);
+module_exit(ecm_exit);
+
+MODULE_LICENSE("GPL");
--- a/drivers/usb/gadget/Makefile
+++ b/drivers/usb/gadget/Makefile
@@ -26,26 +26,28 @@ obj-$(CONFIG_USB_M66592)	+= m66592-udc.o
 gadget_core-objs		:= composite.o usbstring.o config.o \
 					epautoconf.o
 gadget_ether_core-objs		:= u_ether.o
+gadget_ecm_core-objs		:= f_ecm.o
 
 g_zero-objs			:= zero.o f_sourcesink.o f_loopback.o
-g_ether-objs			:= ether.o f_subset.o f_ecm.o
+g_ether-objs			:= ether.o f_subset.o
 g_serial-objs			:= serial.o u_serial.o f_acm.o f_serial.o
 g_midi-objs			:= gmidi.o
 gadgetfs-objs			:= inode.o
 g_file_storage-objs		:= file_storage.o
 g_printer-objs			:= printer.o
-g_cdc-objs			:= cdc2.o f_ecm.o u_serial.o f_acm.o
+g_cdc-objs			:= cdc2.o u_serial.o f_acm.o
 
 ifeq ($(CONFIG_USB_ETH_RNDIS),y)
 	g_ether-objs		+= f_rndis.o rndis.o
 endif
  
 obj-$(CONFIG_USB_ZERO)		+= g_zero.o gadget_core.o
-obj-$(CONFIG_USB_ETH)		+= g_ether.o gadget_core.o gadget_ether_core.o
+obj-$(CONFIG_USB_ETH)		+= g_ether.o gadget_core.o gadget_ether_core.o \
+					gadget_ecm_core.o
 obj-$(CONFIG_USB_GADGETFS)	+= gadgetfs.o
 obj-$(CONFIG_USB_FILE_STORAGE)	+= g_file_storage.o gadget_core.o
 obj-$(CONFIG_USB_G_SERIAL)	+= g_serial.o gadget_core.o
 obj-$(CONFIG_USB_G_PRINTER)	+= g_printer.o gadget_core.o
 obj-$(CONFIG_USB_MIDI_GADGET)	+= g_midi.o gadget_core.o
-obj-$(CONFIG_USB_CDC_COMPOSITE) += g_cdc.o gadget_core.o gadget_ether_core.o
-
+obj-$(CONFIG_USB_CDC_COMPOSITE) += g_cdc.o gadget_core.o gadget_ether_core.o \
+					gadget_ether_core.o
